src/coprocessor/codec/batch/lazy_column.rs

```diff
@@ -19,58 +18,25 @@ use crate::coprocessor::codec::Result;
 /// to avoid unnecessary repeated serialization / deserialization. In future, Coprocessor will
 /// respond all data in Arrow format which is different to the format in storage. At that time,
 /// this structure is no longer useful and should be removed.
+#[derive(Clone, Debug)]
 pub enum LazyBatchColumn {
-    /// Ensure that small datum values (i.e. Int, Real, Time) are stored compactly.
-    /// When using VarInt encoding there are 9 bytes. Plus 1 extra byte to store the datum flag.
-    /// Thus totally there are 10 bytes.
-    Raw(Vec<SmallVec<[u8; 10]>>),
+    Raw(BufferVec),
     Decoded(VectorValue),
 }
 
-impl std::fmt::Debug for LazyBatchColumn {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            LazyBatchColumn::Raw(ref v) => {
-                let vec_display: Vec<_> = v
-                    .iter()
-                    .map(|item| tikv_util::escape(item.as_slice()))
-                    .collect();
-                f.debug_tuple("Raw").field(&vec_display).finish()
-            }
-            LazyBatchColumn::Decoded(ref v) => f.debug_tuple("Decoded").field(v).finish(),
-        }
-    }
-}
-
 impl From<VectorValue> for LazyBatchColumn {
     #[inline]
     fn from(vec: VectorValue) -> Self {
         LazyBatchColumn::Decoded(vec)
     }
 }
 
-impl Clone for LazyBatchColumn {
-    #[inline]
-    fn clone(&self) -> Self {
-        match self {
-            LazyBatchColumn::Raw(v) => {
-                // This is much more efficient than `SmallVec::clone`.
-                let mut raw_vec = Vec::with_capacity(v.capacity());
-                for d in v {
-                    raw_vec.push(SmallVec::from_slice(d.as_slice()));
-                }
-                LazyBatchColumn::Raw(raw_vec)
-            }
-            LazyBatchColumn::Decoded(v) => LazyBatchColumn::Decoded(v.clone()),
-        }
-    }
-}
-
 impl LazyBatchColumn {
     /// Creates a new `LazyBatchColumn::Raw` with specified capacity.
     #[inline]
     pub fn raw_with_capacity(capacity: usize) -> Self {
-        LazyBatchColumn::Raw(Vec::with_capacity(capacity))
+        // 11 = MAX_VAR_INT_LEN(10) + Datum Flag(1)
```

Comment:
Doesn't MAX_VAR_INT_LEN already include the datum byte?

Reply:
Nop. That's the codec part.

For why MAX_VAR_INT_LEN is 10.. 7*9=63, but we need to store 64 bit. Also see https://github.com/facebook/folly/blob/988daad82f08812af4659f4126149b50f8565835/folly/Varint.h#L50

Reply:
prefer to use a constant here